cmd/query/app/query_parser.go

```diff
@@ -143,36 +189,169 @@ func (p *queryParser) parse(r *http.Request) (*traceQueryParameters, error) {
+		return bqp, err
+	}
+	bqp.SpanKinds, err = parseSpanKinds(r, spanKindParam, defaultMetricsSpanKinds)
+	if err != nil {
+		return bqp, err
+	}
+	endTs, err := p.parseTime(r, endTsParam, time.Millisecond)
+	if err != nil {
+		return bqp, err
+	}
+	lookback, err := parseDuration(r, lookbackParam, dp, defaultMetricsQueryLookbackDuration)
+	if err != nil {
+		return bqp, err
+	}
+	step, err := parseDuration(r, stepParam, dp, defaultMetricsQueryStepDuration)
+	if err != nil {
+		return bqp, err
+	}
+	ratePer, err := parseDuration(r, rateParam, dp, defaultMetricsQueryRateDuration)
+	if err != nil {
+		return bqp, err
+	}
+	bqp.EndTime = &endTs
+	bqp.Lookback = &lookback
+	bqp.Step = &step
+	bqp.RatePer = &ratePer
+	return bqp, err
+}
+
+// parseTime parses the time parameter of an HTTP request that is represented the number of "units" since epoch.
+// If the time parameter is empty, the current time will be returned.
+func (p *queryParser) parseTime(r *http.Request, paramName string, units time.Duration) (time.Time, error) {
+	formValue := r.FormValue(paramName)
+	if formValue == "" {
+		if paramName == startTimeParam {
 			return p.timeNow().Add(-1 * p.traceQueryLookbackDuration), nil
 		}
 		return p.timeNow(), nil
 	}
-	micros, err := strconv.ParseInt(value, 10, 64)
+	t, err := strconv.ParseInt(formValue, 10, 64)
 	if err != nil {
-		return time.Time{}, err
+		return time.Time{}, newParseError(err, paramName)
 	}
-	return time.Unix(0, 0).Add(time.Duration(micros) * time.Microsecond), nil
+	return time.Unix(0, 0).Add(time.Duration(t) * units), nil
 }
 
-func (p *queryParser) parseDuration(durationParam string, r *http.Request) (time.Duration, error) {
-	durationInput := r.FormValue(durationParam)
-	if len(durationInput) > 0 {
-		duration, err := time.ParseDuration(durationInput)
-		if err != nil {
-			return 0, fmt.Errorf("cannot not parse %s: %w", durationParam, err)
+// parseDuration parses the duration parameter of an HTTP request using the provided durationParser.
+// If the duration parameter is empty, the given defaultDuration will be returned.
+func parseDuration(r *http.Request, paramName string, dp durationParser, defaultDuration time.Duration) (time.Duration, error) {
+	formValue := r.FormValue(paramName)
+	if formValue == "" {
+		return defaultDuration, nil
+	}
+	d, err := dp.parseDuration(formValue)
+	if err != nil {
+		return 0, newParseError(err, paramName)
+	}
+	return d, nil
+}
+
+func parseBool(r *http.Request, paramName string) (b bool, err error) {
+	formVal := r.FormValue(paramName)
+	if formVal == "" {
+		return false, nil
+	}
+	b, err = strconv.ParseBool(formVal)
+	if err != nil {
+		return b, newParseError(err, paramName)
+	}
+	return b, nil
+}
+
+// parseSpanKindParam parses the input span kinds to filter for in the metrics query.
+// Valid input span kinds are the string representations from the OpenTelemetry model/proto/metrics/otelspankind.proto.
+// For example:
+// - "SPAN_KIND_SERVER"
+// - "SPAN_KIND_CLIENT"
```

Comment:
is this consistent with the strings we used in the Operations API?  I thought we used strings client/server, as defined in OpenTracing.

Reply:
Good pickup, I couldn't find formal docs in comments but found evidence where "server" and "client" are used in the [operations API unit tests](https://github.com/jaegertracing/jaeger/blob/master/cmd/query/app/http_handler_test.go#L581-L582). 

Added a mapping from jaeger/opentracing span kinds to OTEL span kinds.